File: LB-2/Car.cpp
//
// Created by justas327 on 2019-10-07.
//

#include "Car.h"

void Car::fromJson(json data) {
    brand = data["brand"];
    makeYear = data["makeYear"];
    mileage = data["mileage"];
}

int Car::getNumber() {
    int number = ((makeYear + (int) mileage + (int) brand[0]) % 40);
    return fib(number);
}

string Car::toJson() {
    return "{\n"
           "'brand' : '" + brand + "',\n"
                                   "'makeYear' : " + to_string(makeYear) + ",\n"
                                                                           "'mileage' : " + to_string(mileage) +
           "\n"
           "},";
}

int Car::fib(int n) {
    if (n <= 1)
        return n;
    return fib(n - 1) + fib(n - 2);
}

const string &Car::getBrand() const {
    return brand;
}

int Car::getMakeYear() const {
    return makeYear;
}

double Car::getMileage() const {
    return mileage;
}

File: LD-2/car.go
package main

import "fmt"

type Car struct {
	Brand    string
	MakeYear int
	Mileage  float32
}

func (c *Car) toString() string {
	//fmt.Printf("%s %d %f\n", c.Brand, c.MakeYear, c.Mileage)
	return fmt.Sprintf("|%20v|%20v|%20v|%20v|", c.Brand, c.MakeYear, c.Mileage, c.getNumber())
}

func (c *Car) getNumber() int {
	number := c.MakeYear + int(c.Mileage) + int(c.Brand[0])
	return c.fibonacci(number % 40)
}

func (c *Car) fibonacci(n int) int {
	if n <= 1 {
		return n
	}

	return c.fibonacci(n-1) + c.fibonacci(n-2)
}
